glean -install/update show panic

while using glean -install/update, log shows:
`
panic: exit status 1

goroutine 1 [running]:
github.com/alissa-tung/glean/glean.InstallLean()
        /Users/alissa/pod/glean/glean/lean.go:80 +0x932
main.main()
        /Users/alissa/pod/glean/main.go:21 +0x8f
`
Is this a bug?

--- a/glean/config.go
+++ b/glean/config.go
@@ -9,7 +9,7 @@ import (
 )
 
 const (
-	gleanVersion = "v0.1.17"
+	gleanVersion = "v0.1.19"
 
 	urlBase = "https://mirror.sjtu.edu.cn/elan"
 )

--- a/glean/lake.go
+++ b/glean/lake.go
@@ -149,7 +149,6 @@ func LakeSyncPackages() {
 }
 
 func FetchProofWidgetsRelease(version string, path string) {
-	// resourceUrl := urlBase + "/proofwidgets/releases/download/" + version + "/ProofWidgets4.tar.gz"
 	resourceUrl := urlBase + "/proofwidgets/releases/download/" + version + "?mirror_intel_list"
 	fmt.Println("Fetching from " + resourceUrl)
 	response, err := http.Get(resourceUrl)

--- a/glean/lake.go
+++ b/glean/lake.go
@@ -149,12 +149,27 @@ func LakeSyncPackages() {
 }
 
 func FetchProofWidgetsRelease(version string, path string) {
-	resourceUrl := urlBase + "/proofwidgets/releases/download/" + version + "/ProofWidgets4.tar.gz"
+	// resourceUrl := urlBase + "/proofwidgets/releases/download/" + version + "/ProofWidgets4.tar.gz"
+	resourceUrl := urlBase + "/proofwidgets/releases/download/" + version + "?mirror_intel_list"
 	fmt.Println("Fetching from " + resourceUrl)
 	response, err := http.Get(resourceUrl)
 	if err != nil {
 		panic("http.Get error: " + err.Error() + ", resourceUrl = `" + resourceUrl + "`")
 	}
+	redirectUrl := response.Request.URL.String()
+	finalUrl := strings.Replace(redirectUrl, "mirror_clone_list.html", "ProofWidgets4.tar.gz", 1)
+	log.Println("will get `" + finalUrl + "`")
+	response, err = http.Get(finalUrl)
+	if err != nil {
+		panic("http.Get error: " + err.Error() + ", resourceUrl = `" + finalUrl + "`")
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			return
+		}
+	}(response.Body)
+
 	if response.StatusCode != http.StatusOK {
 		panic("http.Get error")
 	}

--- a/glean/lean.go
+++ b/glean/lean.go
@@ -18,15 +18,29 @@ func InstallLean() {
 	if strings.Contains(*version, "nightly") {
 		nightlyIndex := strings.Index(*version, "nightly")
 		nightlyVersion := *version
-		resourceUrl = urlBase + "/leanprover/lean4_nightly/releases/download/" + nightlyVersion[nightlyIndex:] + "/" + releaseName
+		resourceUrl = urlBase + "/leanprover/lean4_nightly/releases/download/" + nightlyVersion[nightlyIndex:] + "?mirror_intel_list"
 	} else {
-		resourceUrl = urlBase + "/leanprover/lean4/releases/download/v" + *version + "/" + releaseName
+		resourceUrl = urlBase + "/leanprover/lean4/releases/download/v" + *version + "?mirror_intel_list"
 	}
 	log.Println("will get `" + resourceUrl + "`")
 
 	response, err := http.Get(resourceUrl)
 	if err != nil {
 		panic("http.Get error: " + err.Error() + ", resourceUrl = `" + resourceUrl + "`")
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			return
+		}
+	}(response.Body)
+
+	redirectUrl := response.Request.URL.String()
+	finalUrl := strings.Replace(redirectUrl, "mirror_clone_list.html", releaseName, 1)
+	log.Println("will get `" + finalUrl + "`")
+	response, err = http.Get(finalUrl)
+	if err != nil {
+		panic("http.Get error: " + err.Error() + ", resourceUrl = `" + finalUrl + "`")
 	}
 	defer func(Body io.ReadCloser) {
 		err := Body.Close()

--- a/glean/update.go
+++ b/glean/update.go
@@ -8,7 +8,6 @@ import (
 	"io/ioutil"
 	"log"
 	"net/http"
-	"net/url"
 	"os"
 	"os/exec"
 	"path/filepath"
@@ -70,10 +69,24 @@ func CheckUpdate() {
 	fmt.Println("New version available:", latestVersion)
 
 	releaseName := buildGleanReleaseName()
-	releaseUrl, err := url.JoinPath(urlBase, "glean", "releases", "download", latestVersion, releaseName)
+	releaseUrl := urlBase + "/glean/releases/download/" + latestVersion + "?mirror_intel_list"
 	response, err := http.Get(releaseUrl)
 	if err != nil {
 		panic(err)
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			return
+		}
+	}(response.Body)
+
+	redirectUrl := response.Request.URL.String()
+	finalUrl := strings.Replace(redirectUrl, "mirror_clone_list.html", releaseName, 1)
+	log.Println("will get `" + finalUrl + "`")
+	response, err = http.Get(finalUrl)
+	if err != nil {
+		panic("http.Get error: " + err.Error() + ", resourceUrl = `" + finalUrl + "`")
 	}
 	defer func(Body io.ReadCloser) {
 		err := Body.Close()

